pkg/client/mapiserver/snapshot.go

```diff
@@ -0,0 +1,202 @@
+/*
+Copyright 2017 The OpenEBS Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package mapiserver
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"time"
+
+	client "github.com/openebs/maya/pkg/client/jiva"
+	"github.com/openebs/maya/types/v1"
+	yaml "gopkg.in/yaml.v2"
+)
+
+const (
+	http_timeout = 5 * time.Second
+)
+
+// CreateSnapshot creates a snapshot of volume by invoking the API call to m-apiserver
+func CreateSnapshot(volName string, snapName string) error {
+
+	_, err := GetStatus()
+	if err != nil {
+		err := errors.New(fmt.Sprintf("Unable to contact maya-apiserver: %s", GetURL()))
+		return err
+	}
+
+	var snap v1.SnapshotAPISpec
+
+	snap.Kind = "VolumeSnapshot"
+	snap.APIVersion = "v1"
+	snap.Metadata.Name = snapName
+	snap.Spec.VolumeName = volName
+
+	//Marshal serializes the value provided into a YAML document
+	yamlValue, _ := yaml.Marshal(snap)
+
+	fmt.Printf("Volume snapshot spec created:\n%v\n", string(yamlValue))
+
+	url := GetURL() + "/latest/snapshots/create/"
+	req, err := http.NewRequest("POST", url, bytes.NewBuffer(yamlValue))
+	if err != nil {
+		return err
+	}
+
+	req.Header.Add("Content-Type", "application/yaml")
```

Comment:
This need not be yaml if struct can be send. I feel sending yamls from within a CLI is time consuming & un-necessary.